Sources/ProjectDescription/Headers.swift

```diff
@@ -82,4 +88,48 @@ public struct Headers: Codable, Equatable {
             exclusionRule: exclusionRule
         )
     }
+
+    public static func headers(from list: FileList,
+                               umbrella: Path,
+                               private: FileList? = nil,
+                               allOthersAsProject: Bool) -> Headers
+    {
+        return .init(
+            public: list,
+            umbrellaHeader: umbrella,
+            private: `private`,
+            project: allOthersAsProject ? list : nil,
+            exclusionRule: .projectExcludesPrivateAndPublic
+        )
+    }
+
+    /// all left found headers from the list
+    /// (after public/private scopes loading)
+    /// will be added with `project` visibility
+    public static func allHeaders(from list: FileList,
```

Comment:
I would improve this doc and the one of the init below by specifying:
- what `from` is (I guess the list of all headers?)
- how does `umbrella` impacts the public/project (I guess the ones found in the umbrella will be public)?

Reply:
updated